src/ui/atoms/toast/Toast.tsx

```diff
@@ -30,48 +31,79 @@ type ToastProps = Readonly<{
    */
   readonly onOpenChange?: (isOpened: boolean) => void
+  /**
+   * Prevents the toast from closing automatically.
+   */
+  readonly preventAutoClose?: boolean
 }>
 
+// eslint-disable-next-line max-lines-per-function
 export const Toast: React.FC<ToastProps> = ({
   isOpened,
   autoDuration = 3000,
   severityLevel,
   title,
   description,
-  onOpenChange
+  onOpenChange,
+  preventAutoClose
 }: DeepReadonly<ToastProps>) => {
+  const closeButton = (
+    <ToastPrimitive.Close className="okp4-toast-close">
+      <Icon name="close" size={20} />
+    </ToastPrimitive.Close>
+  )
+
+  const renderTitle = (): string | JSX.Element | undefined =>
+    title && (
+      <ToastPrimitive.Title asChild>
+        <Typography color="highlighted-text" fontSize="small" fontWeight="bold">
+          {title}
+        </Typography>
+      </ToastPrimitive.Title>
+    )
+
+  const renderDescription = (): string | JSX.Element | undefined =>
+    description && (
+      <ToastPrimitive.Description asChild>
+        {isString(description) ? (
+          <Typography color="highlighted-text" fontSize="small" fontWeight="light">
+            {description}
+          </Typography>
+        ) : (
+          description
+        )}
+      </ToastPrimitive.Description>
+    )
+
   return (
     <ToastPrimitive.Provider swipeDirection="right">
       <ToastPrimitive.Root
         className={`okp4-toast-root ${severityLevel}`}
         defaultOpen={false}
-        duration={autoDuration}
+        duration={preventAutoClose ? Infinity : autoDuration}
         onOpenChange={onOpenChange}
         open={isOpened}
       >
-        {title && (
-          <ToastPrimitive.Title asChild className="okp4-toast-title">
-            <Typography
-              as="span"
-              color="highlighted-text"
-              fontSize="small"
-              fontWeight="bold"
-              noWrap
-            >
-              {title}
-            </Typography>
-          </ToastPrimitive.Title>
+        {title && preventAutoClose ? (
+          <div className="okp4-toast-wrapper">
+            {renderTitle()}
+            {closeButton}
+          </div>
+        ) : (
+          renderTitle()
         )}
-        {description && (
-          <ToastPrimitive.Description asChild className="okp4-toast-description">
-            {isString(description) ? (
-              <Typography as="span" color="highlighted-text" fontSize="small" fontWeight="light">
-                {description}
-              </Typography>
-            ) : (
-              description
-            )}
-          </ToastPrimitive.Description>
+
+        {description && preventAutoClose ? (
+          title ? (
+            renderDescription()
+          ) : (
+            <div className="okp4-toast-wrapper">
+              {renderDescription()}
+              {closeButton}
+            </div>
+          )
+        ) : (
+          renderDescription()
```

Comment:
Please, make readable rendering functions instead of nested ternaries that ruin the understanding of the jsx part.
There are too much assertions on properties everywhere, we need to clean that before merging